etna/transforms/embeddings/models/ts2vec.py

```diff
@@ -0,0 +1,250 @@
+        n_iters:
+            The number of iterations. When this reaches, the training stops. If both n_epochs and n_iters are not specified,
+            a default setting would be used that sets n_iters to 200 for a dataset with size <= 100000, 600 otherwise.
+        verbose:
+            Whether to print the training loss after each epoch.
+        causal:
+            When this param is set to True, the future information would not be encoded into representation of each timestamp.
+        mask:
+            The mask used by encoder on the test phase can be specified with this parameter. The possible options are:
+                * 'binomial' - mask timestamp with probability 0.5 (default one, used in the paper). It is used on the training phase.
+                * 'continuous' - mask random windows of timestamps
+                * 'all_true' - mask none of the timestamps
+                * 'all_false' - mask all timestamps
+                * 'mask_last' - mask last timestamp
+        max_train_length:
+            The maximum allowed sequence length for training. For sequence with a length greater than ``max_train_length``,
+            it would be cropped into some sequences, each of which has a length less than ``max_train_length``.
+        temporal_unit:
+            The minimum unit to perform temporal contrast. When training on a very long sequence,
+            this param helps to reduce the cost of time and memory.
+        Notes
+        -----
+        In case of long series to reduce memory consumption it is recommended to use max_train_length parameter or manually break the series into smaller subseries (see ETNA-2185).
+        """
+        super().__init__(output_dims=output_dims)
+        self.input_dims = input_dims
+        self.output_dims = output_dims
+        self.hidden_dims = hidden_dims
+        self.depth = depth
+        self.encoding_window = encoding_window
+        self.max_train_length = max_train_length
+        self.temporal_unit = temporal_unit
+
+        self.device = device
+        self.lr = lr
+        self.batch_size = batch_size
+
+        # Train params
+        self.n_epochs = n_epochs
+        self.n_iters = n_iters
+        self.verbose = verbose
+
+        # Encode params
+        self.mask = mask
+        self.causal = causal
+        self.sliding_length = sliding_length
+        self.sliding_padding = sliding_padding
+
+        self.embedding_model = TS2Vec(
+            input_dims=self.input_dims,
+            output_dims=self.output_dims,
+            hidden_dims=self.hidden_dims,
+            depth=self.depth,
+            max_train_length=self.max_train_length,
+            temporal_unit=self.temporal_unit,
+            device=self.device,
+            lr=self.lr,
+            batch_size=self.batch_size,
+        )
+
+    def _prepare_data(self, df: pd.DataFrame) -> np.ndarray:
+        """Prepare data for the embedding model."""
```

Comment:
We should probably clarify what this transformation does. E.g. in the comments or in the docstring.